app/lua53/host/luac.c

```diff
@@ -72,282 +69,195 @@ static void usage(const char *message) {
 }
 
+#include "lauxlib.h"
+static int execfile(lua_State *L, const char *name);
 
 static int pmain(lua_State *L) {
   int argc = (int) lua_tointeger(L, 1);
-  char **argv = (char **) lua_touserdata(L, 2);
-  char **filelist = alloca(argc * sizeof(char *));
-  const Proto *f;
-  int i, status;
-  if (!lua_checkstack(L, argc + 1))
-    fatal("too many input files");
-  if (execute || address) {
-    luaL_openlibs(L);  /* the nodemcu open will throw to signal an LFS reload */
-    status = dofile(L, execute);
-    if (status != LUA_OK)
-      return 0;
-  }
-  if (argc == 0)
-    return 0;
-  reorderfiles(L, argc, filelist, argv);
+  const char **argv = (const char ** ) lua_touserdata(L, 2);
+  int i;
+  luaL_openlibs(L);
+  if (!lua_checkstack(L, argc + 1)) fatal("too many input files");
+  if (execute && (execfile(L, execute) != LUA_OK))
+    lua_error(L);
+  lua_createtable(L, argc, 0);
   for (i = 0; i < argc; i++) {
-    const char *filename = IS("-") ? NULL : filelist[i];
-    if (luaL_loadfile(L, filename) != LUA_OK)
+    size_t l;   
+    lua_pushstring(L, IS("-") ?  "stdin" : argv[i]);
+    const char *p = corename(L, tsvalue(L->top-1), &l);
+    lua_pushlstring(L, p, l);
+    lua_remove(L, -2);
+    if (luaL_loadfile(L, IS("-") ? NULL : argv[i]) != LUA_OK)
       fatal(lua_tostring(L, -1));
-//TODO: if strip = 2, replace proto->source by basename
+    lua_rawset(L, -3);
   }
-  f = combine(L, argc + (execute ? 1 : 0), lookup);
-  if (listing) luaU_print(f, listing > 1);
-  if (dumping) {
-    int result;
+//  if (listing) luaU_print(f, listing > 1);
+  (void)(PrintFunction);
+  if (dumping  && argc > 0) {
     FILE *D = (output == NULL) ? stdout : fopen(output, "wb");
     if (D == NULL) cannot("open");
     lua_lock(L);
-    if (flash) {
-      UNUSED(address);
-      UNUSED(maxSize);
-      result = luaU_DumpAllProtos(L, f, writer, &D, stripping);
-    } else {
-      result = luaU_dump(L, f, writer, cast(void *, &D), stripping);
-    }
+    luaU_dump(L, hvalue(L->top-1), writer, D, stripping);
     lua_unlock(L);
-    if (result == LUA_ERR_CC_INTOVERFLOW)
-      fatal("value too big or small for target integer type");
-    if (result == LUA_ERR_CC_NOTINTEGER)
-      fatal("target lua_Number is integral but fractional value found");
     if (ferror(D)) cannot("write");
     if (fclose(D)) cannot("close");
   }
   return 0;
 }
 
+int request_restart;
+
 int main(int argc, char *argv[]) {
   lua_State *L;
   int i = doargs(argc, argv);
-  int j, status;
-  argc -= i; argv += i;
+  int status;
+  const char *err;
+  argc -= i;
+  argv += i;
   if (argc <= 0 && execute == 0 && address == 0) usage("no input files given");
-  if (address)
-    luaN_setabsolute(address);
-  for (j = 0; j < 2 ; j++) {
+  //if (address)
+  //  luaN_setabsolute(address);
+
+  do {
+    request_restart = 0;
     L = luaL_newstate();
-    if (L == NULL) fatal("not enough memory for state");
+    if (L == NULL)
+      fatal("not enough memory for state");
```

Comment:
There are a great many "stylistic" changes like the above three lines of diff in these commits.  Do they increase or decrease the diff between "upstream Lua" and "nodemcu Lua"?  Most of them appear to my mind to be improvements in the code readability; if they increase the gap between "upstream" and "nodemcu", can we send them upstream? :)

Reply:
`luac.c` needs a lot more work largely because the unification of the LFS  and LC formats.  Given the volume of change and the previous nasty non-standard formatting of lauc.c which is nothing like the rest of the lua core codebase, I have been considering running it through a pretty print style formatter to make it more readable anyway.